src/html5/parser/document.rs

```diff
@@ -692,6 +698,59 @@ impl DocumentHandle {
 
         Ok(())
     }
+
+    /// Perform a single query against the document.
+    /// If query search type is uninitialized, returns an error.
+    /// Otherwise, returns a vector of NodeIds that match the predicate in tree order.
```

Comment:
After thinking about this before bed.. this implementation is technically not correct. According to DOM spec, tree order is depth-first which this current implementation does not (always) do. Will need to be updated

Reply:
Fixed algo to search depth-first instead of breadth-first